extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryKit.java

```diff
@@ -178,6 +183,11 @@ public QueryDefinition makeQueryDefinition(
       }
     }
 
+    // If window has an empty over, we want a single worker to process entire data for window function evaluation.
+    if (windowHasEmptyOver) {
+      scanShuffleSpec = MixShuffleSpec.instance();
```

Comment:
Setting the shuffle spec like this ignores the cluster by columns that were added previous to the window functions piece of code. This means that we do not have the sorting according to those columns, is this correct?

Reply:
I think so. WindowOperatorQueryKit would be adding a sorting operator anyway (if it needs to), so we just need a MixShuffleSpec to bring all the data into a single worker for window function evaluation.